src/transformers/models/oneformer/processing_oneformer.py

```diff
@@ -37,34 +70,43 @@ class OneFormerProcessor(ProcessorMixin):
             The image processor is a required input.
         tokenizer ([`CLIPTokenizer`, `CLIPTokenizerFast`]):
             The tokenizer is a required input.
-        max_seq_len (`int`, *optional*, defaults to 77)):
+        max_seq_length (`int`, *optional*, defaults to 77):
             Sequence length for input text list.
-        task_seq_len (`int`, *optional*, defaults to 77):
+        task_seq_length (`int`, *optional*, defaults to 77):
             Sequence length for input task token.
     """
 
     attributes = ["image_processor", "tokenizer"]
     image_processor_class = "OneFormerImageProcessor"
     tokenizer_class = ("CLIPTokenizer", "CLIPTokenizerFast")
+    optional_call_args = ["task_inputs", "segmentation_maps"]
 
     def __init__(
-        self, image_processor=None, tokenizer=None, max_seq_length: int = 77, task_seq_length: int = 77, **kwargs
+        self,
+        image_processor=None,
+        tokenizer=None,
     ):
         if image_processor is None:
             raise ValueError("You need to specify an `image_processor`.")
         if tokenizer is None:
             raise ValueError("You need to specify a `tokenizer`.")
 
-        self.max_seq_length = max_seq_length
-        self.task_seq_length = task_seq_length
-
         super().__init__(image_processor, tokenizer)
 
-    def _preprocess_text(self, text_list=None, max_length=77):
-        if text_list is None:
-            raise ValueError("tokens cannot be None.")
-
-        tokens = self.tokenizer(text_list, padding="max_length", max_length=max_length, truncation=True)
+    def _preprocess_text(
+        self,
+        text_list: PreTokenizedInput,
+        max_length: Optional[int] = None,
+        text_kwargs: Optional[dict] = None,
+    ):
+        if text_kwargs is None:
+            text_kwargs = {}
+        tokens = self.tokenizer(
+            text_list,
+            max_length=max_length if max_length is not None else text_kwargs.get("max_length", 77),
+            padding=text_kwargs.get("padding", "max_length"),
+            truncation=text_kwargs.get("truncation", True),
```

Comment:
No need to redefine the defaults kwargs here in the get, as if they are not specified, they will necessarily be "max_length" and True respectively, you can just have:
```suggestion
            padding=text_kwargs.get("padding"),
            truncation=text_kwargs.get("truncation"),
```

Reply:
I believe this is needed because `encode_inputs` does not pass in any dict, so `text_kwargs` will become `{}`, without default.

Reply:
what is weird for me here is taht we don't use all the textkwargs. Why not just pass `**text_kwargs`? They can be typed as well (maybe just tokenizer kwargs) and you just update them with the args passed!